src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -288,16 +288,18 @@ public sealed class SecurityIdentifier : IdentityReference, IComparable<Security
         //
 
         private IdentifierAuthority _identifierAuthority;
-        private int[] _subAuthorities;
-        private byte[] _binaryForm;
-        private SecurityIdentifier _accountDomainSid;
+        // values guaranteed to be non-null on account of CreateFromParts()
+        // method called by every constructor in this class.
+        private int[] _subAuthorities = null!;
+        private byte[] _binaryForm = null!;
```

Comment:
I'm not quite sure what this means, the comment indicates they're non-null typed but then it's changed change explicitly applies null to them? Why are the nulls needed?

Reply:
Because the values are initialized by a method and not the constructor directly, the analyzer will complain about nullability unless overriden like above.

Reply:
So we'd need a way to express to the analyzer that the called method guarnetees that these non-argument members are non-null as a postcondition, which isn't currently possible. Or we could change create parts to take the assignment to the caller not the calle so it can see the assignment, i don't like the idea of rearranging code just to work around an analyzers limitation though.

Whether this is a (really tiny) problem depends on which is more important, cleaner code without null assignments or getting a clean slate on the analyzer.